package/src/components/Message/hooks/useProcessReactions.ts

```diff
@@ -0,0 +1,113 @@
+import { ComponentType, useMemo } from 'react';
+
+import { ReactionResponse } from 'stream-chat';
+
+import {
+  MessagesContextValue,
+  useMessagesContext,
+} from '../../../contexts/messagesContext/MessagesContext';
+import { DefaultStreamChatGenerics } from '../../../types/types';
+import { ReactionData } from '../../../utils/utils';
+import { ReactionListProps } from '../MessageSimple/ReactionList';
+
+export type ReactionSummary = {
+  count: number;
+  firstReactionAt: Date | null;
+  Icon: ComponentType | null;
+  isOwnReaction: boolean;
+  lastReactionAt: Date | null;
+  latestReactedUserNames: string[];
+  type: string;
+  unlistedReactedUserCount: number;
+};
+
+export type ReactionsComparator = (a: ReactionSummary, b: ReactionSummary) => number;
+
+type UseProcessReactionsParams<
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
+> = Pick<
+  ReactionListProps<StreamChatGenerics>,
+  'own_reactions' | 'reaction_groups' | 'latest_reactions'
+> &
+  Partial<Pick<MessagesContextValue<StreamChatGenerics>, 'supportedReactions'>> & {
+    sortReactions?: ReactionsComparator;
+  };
+
+export const defaultReactionsSort: ReactionsComparator = (a, b) => {
+  if (a.firstReactionAt && b.firstReactionAt) {
+    return +a.firstReactionAt - +b.firstReactionAt;
+  }
+
+  return a.type.localeCompare(b.type, 'en');
+};
+
+const isOwnReaction = <
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
+>(
+  reactionType: string,
+  ownReactions?: ReactionResponse<StreamChatGenerics>[] | null,
+) => (ownReactions ? ownReactions.some((reaction) => reaction.type === reactionType) : false);
+
+const isSupportedReaction = (reactionType: string, supportedReactions: ReactionData[]) =>
+  supportedReactions
+    ? supportedReactions.some((reactionOption) => reactionOption.type === reactionType)
+    : false;
+
+const getEmojiByReactionType = (reactionType: string, supportedReactions: ReactionData[]) =>
+  supportedReactions.find(({ type }) => type === reactionType)?.Icon ?? null;
+
+const getLatestReactedUserNames = (reactionType: string, latestReactions?: ReactionResponse[]) =>
+  latestReactions
+    ? latestReactions.flatMap((reaction) => {
+        if (reactionType && reactionType === reaction.type) {
+          const username = reaction.user?.name || reaction.user?.id;
+          return username ? [username] : [];
+        }
+        return [];
+      })
+    : [];
+
+export const useProcessReactions = <
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
```

Comment:
can you add a small documentation of this hook of what it does please.. useful for future maintenance

Reply:
Done